src/RA_Defs.h

```diff
@@ -225,30 +255,382 @@ typedef DWORD			ARGB;
+    return static_cast<_CSTD DWORD>(n);
+} // end operator""_dw
+
+// streamsize varies as well
+/// <summary>
+/// A literal
+/// </summary>
+/// <param name="n">The n.</param>
+/// <returns></returns>
+_CONSTANT_VAR operator""_ss(unsigned long long n) noexcept {
+    return static_cast<std::streamsize>(n);
+} // end operator""_ss
+
+// Literal for unsigned short
+_CONSTANT_VAR operator""_hu(unsigned long long n) noexcept {
+    return static_cast<std::uint16_t>(n);
+} // end operator""_hu
+
+
+_CONSTANT_VAR operator""_dp(unsigned long long n) noexcept {
+    return static_cast<DataPos>(n);
+} // end operator""_dp
+
+// If you follow the standard every alias is considered mutually exclusive, if not don't worry about it
+// These are here if you follow the standards (starts with ISO C++11/C11) rvalue conversions
+
+
+_CONSTANT_VAR operator""_ba(unsigned long long n) noexcept
+{
+    return static_cast<ByteAddress>(n);
+} // end operator""_ba
+
+
+_CONSTANT_VAR operator""_achid(unsigned long long n) noexcept {
+    return static_cast<AchievementID>(n);
+} // end operator""_achid
+
+
+_CONSTANT_VAR operator""_lbid(unsigned long long n) noexcept {
+    return static_cast<LeaderboardID>(n);
+} // end operator""_lbid
+
+
+_CONSTANT_VAR operator""_gameid(unsigned long long n) noexcept {
+    return static_cast<GameID>(n);
+} // end operator""_gameid
+
+} // inline namespace int_literals
+
+
+inline namespace conversions {
+
+// Noticed some weird uses of rvalues returning as lvalues, whatever they aren't used anywhere
+
+std::string DataStreamAsString(const DataStream& stream);
+std::string Narrow(const std::wstring& wstr);
+std::string Narrow(const wchar_t* wstr);
+std::wstring Widen(const std::string& str);
+std::wstring Widen(const char* str);
+
+
+
+
+
+//	No-ops to help convert:
+std::wstring Widen(const wchar_t* wstr);
+std::wstring Widen(const std::wstring& wstr);
+std::string Narrow(const char* str);
+std::string Narrow(const std::string& wstr);
+std::string ByteAddressToString(ByteAddress nAddr);
+
+
+template<typename Enum, typename = std::enable_if_t<std::is_enum_v<Enum>>>
+_CONSTANT_VAR etoi(Enum e) -> std::underlying_type_t<Enum>
+{
+    return static_cast<std::underlying_type_t<Enum>>(e);
+}
+
+// function alias template
+template<typename Enum>
+_CONSTANT_VAR to_integral = etoi<Enum>;
+
+/// <summary>
+///   Converts a standard string with a signed character type to a
+///   <c>DataStream</c>.
+/// </summary>
+/// <param name="str">The string.</param>
+/// <typeparam name="CharT">
+///   The character type, must be signed or you will get an intellisense error.
+/// </typeparam>
+/// <returns><paramref name="str" /> as a <c>DataStream</c>.</returns>
+/// <remarks>
+///   <c>DataStream</c> is just an alias for an unsigned standard string.
+/// </remarks>
+template<
+    typename CharT,
+    class = std::enable_if_t<is_char_v<CharT> && std::is_signed_v<CharT>>
+>
+DataStream stodata_stream(const std::basic_string<CharT>& str) noexcept
```

Comment:
This should just be `to_data_stream`.

Is it even used?

Reply:
It's preparation, I remember using it in my RASuite fork but it's not used here yet.